tests/test_dsl.py

```diff
@@ -197,3 +198,20 @@ def test_jeremy(self):
         [sum_{} [sum_{X,W,D,Z,Y} P(X,W,D,Z,Y,V)]]]]/[ sum_{Y}[sum_{D,Z,V} [sum_{} [sum_{X,W,Z,Y,V} P(X,W,D,Z,Y,V)]]
         [sum_{}P(Z|D,V)][sum_{} [sum_{X} P(Y|X,D,V,Z,W)P(X|)]][sum_{} [sum_{X,W,D,Z,Y} P(X,W,D,Z,Y,V)]]]]
         '''
+
+
+class TestGetVariables(unittest.TestCase):
+    """Test getting variables."""
+
+    def assert_has_variables(self, expression: Expression, variables: Set[str]) -> None:
+        """Assert the variables are the result of getting variables from the expression."""
+        self.assertEqual(variables, get_variables(expression))
+
+    def test_api(self):
+        """Test the high-level API for getting variables."""
+        for expression, variables in [
+            (P(A @ B), {'B'}),
+            (P(A @ ~B), {'B'}),
```

Comment:
@djinnome this is the best place to add more high-level tests. Since it's the case that we're actually going to want to extract the string variable names before using in the `is_identifiable()` function later, I've written the tests on this level of abstraction